objects/ohitbox/Step_2.gml

```diff
@@ -22,228 +22,400 @@ if (!global.game_paused)
+					}
+				
+					//Allow Cancelling
+					owner.cancelable = true;
+				
+					// Depth Sorting
+					owner.depth = -1;
+					collision_list[| i].owner.depth = 0;
+				
+					// Apply proper knockback direction to Destructable Objects
+					if (collision_list[| i].owner.isDestructibleObject)
 					{
-						var counterParticle = instance_create_layer(global.camObj.x - 80, 0, "Particles", oParticles);
-						with (counterParticle) 
-						{
-							lifetime = global.hitstop;
-							sprite_index = sCOUNTERtext;
-							depth = 2;
-						}
+						collision_list[| i].owner.knockbackDirection = sign(image_xscale);
+					}
+
+					// Reset Frame Advantage Counter
+					oGameManager.frameAdvantage = 0;
+				
+						//Draw hit effect
+					var particle = instance_create_layer(x + (attackProperty.particlexOffset[hitboxID] * owner.image_xscale), y - attackProperty.particleyOffset[hitboxID], "Particles", oParticles);
+					with (particle) 
+					{
+						sprite_index = other.attackProperty.particleEffect[other.hitboxID];
+						image_xscale = sign(other.owner.image_xscale);
+						lifetime = other.attackProperty.particleDuration[other.hitboxID];
 					}
-					
-				} 
-				else 
-				{ // on Regular Hit
-					ProcessHit(attackProperty, collision_list[| i]);
 				}
+			}
+		}
+	}
+}
+else
+{
+	var collisionCheck = place_meeting(x,y, oPlayerHurtbox);
+	var collisionID = noone;
+
+	if (collisionCheck)
+	{
+		// Creates a list containing all of the hurtboxes we're colliding with.
+		var collision_list = ds_list_create();
+		collisionID = instance_place_list(x,y, oPlayerHurtbox, collision_list, false);
+	
+		// If the hurtbox we're touching isn't our own AND that hitbox hasn't hit anything
+		// Iterates over every hurtbox being collided with to check if it can be hit.
+		for (var i = 0; i < collisionID; i++;) {
+		
+			// This code handles multiple hitboxes being used
+			// It checks to see if the ID of this hitbox is contained within the hitByGroup list of the victim.
+			// Whenever a hitbox connects, it adds its ID to the hitByGroup list to the victim
+			var gotHitBy = ds_list_find_index(collision_list[| i].owner.hitByGroup, attackProperty.group[hitboxID])
+			if (collision_list[| i].owner != owner.playerOwner && !hasHit && gotHitBy == -1 && !collision_list[| i].owner.invincible) 
+			{
+				//Set who the player is currently targeting
+				owner.playerOwner.target = collision_list[| i].owner.id;
+			
+				// Projectiles cannot be grabs or throws, so the code for them is ommitted here
+
+				// BLOCKING                                  Check to see if we are blocking correctly
+				if (collision_list[| i].owner.canBlock) && ((attackProperty.attackType[hitboxID] == eAttackType.LOW && collision_list[| i].owner.verticalMoveDir == -1) || attackProperty.attackType[hitboxID] == eAttackType.MID || (attackProperty.attackType[hitboxID] == eAttackType.HIGH && collision_list[| i].owner.verticalMoveDir != -1)){

```

Comment:
This line is super long, break it down like you did above on line 73. Also the { needs to go on its own line.